### PR TITLE
[build] Replace 'go-get' with 'go-install'

### PR DESCRIPTION
**Pull request by alinaryan**
This change is needed because the 'make bundle' command was failing since
'go get' can no longer install executables. Effective in Go 1.17, installing
executables must be done with 'go install'.
Doc Ref: https://go.dev/doc/go-get-install-deprecation

Also upgraded kustomize version to be able to install it with 'go install'

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -114,22 +114,22 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
...
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+	$(call go-install-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
 
 KUSTOMIZE = $(shell pwd)/bin/kustomize
 kustomize: ## Download kustomize locally if necessary.
-	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
+	$(call go-install-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.5.4)
 
-# go-get-tool will 'go get' any package $2 and install it to $1.
+# go-install-tool will 'go get' any package $2 and install it to $1.
```
Comment (saifshaikh48) [inline Makefile:123]: ` ...will 'go get' any package...` -->  `...will 'go install' any package`

File: Makefile [hunk 2]
```diff
@@ -114,22 +114,22 @@ undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/confi
 
 CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
 controller-gen: ## Download controller-gen locally if necessary.
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+	$(call go-install-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
 
 KUSTOMIZE = $(shell pwd)/bin/kustomize
 kustomize: ## Download kustomize locally if necessary.
-	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
+	$(call go-install-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.5.4)
```
Comment (sebsoto) [inline Makefile:121]: theres 2 unrelated changes in here, if the kustomize bump is required, please separate into 2 commits and reframe the PR to reflect the purpose of the changes.

**Review by aravindhp — APPROVED** [review]
Thanks for fixing this @alinaryan
